src/domains/transaction/components/TransactionDetail/TransactionDetails/TransactionDetails.tsx

```diff
@@ -17,13 +17,22 @@ export const TransactionDetails = ({
 	const format = useTimeFormat();
 
 	const timestamp = transaction.timestamp();
-	const data = transaction.data().data;
-
+	console.log(transaction.blockId, transaction.blockId());
 	const { blockHeight } = useBlockHeight({
 		blockId: transaction.blockId(),
 		network: transaction.wallet().network(),
 	});
 
+	const nonce = useCallback(() => {
+		try {
+			const data = transaction.data().data;
+			const nonceValue = (data instanceof Function ? data() : data)?.nonce;
```

Comment:
isn't nonce always available at this point? it's supposed to be set when the transaction is created, so it should be something we always know 🤔